### ISSUE TITLE
Support for client cerificate authentication

### ISSUE THREAD

**Issue by forstern** [yuezk/GlobalProtect-openconnect#21]
Does the client currently support authentication using a client certificate in order to verify the clients authenticity? Is it possible to specify a certain certificate that is used during authentication?

**Comment by yuezk** [comment]
Honestly, I'm not sure about that. If OpenConnect support it, then we can consider adding it into this GUI client.

**Comment by petobens** [comment]
Not sure if it is related but I'm unable to connect and getting the following error:
```2020-09-22 17:13:43.621 INFO  [24307] [GPClient::onVPNLogAvailable@440] Openconnect started successfully, PID=32068
2020-09-22 17:13:43.641 INFO  [24307] [GPClient::onVPNLogAvailable@440] POST https://arpd-gateway-01.networking-is.net/ssl-vpn/getconfig.esp

2020-09-22 17:13:43.752 INFO  [24307] [GPClient::onVPNLogAvailable@440] Connected to 190.x.x.x:443

2020-09-22 17:13:43.859 INFO  [24307] [GPClient::onVPNLogAvailable@440] SSL negotiation with xx.xx.net

2020-09-22 17:13:43.899 INFO  [24307] [GPClient::onVPNLogAvailable@440] Server certificate verify failed: certificate does not match hostname

2020-09-22 17:13:43.899 INFO  [24307] [GPClient::onVPNLogAvailable@440]
Certificate from VPN server "xx.xx.net" failed verification.
Reason: certificate does not match hostname
To trust this server in future, perhaps add this to your command line:
    --servercert pin-sha256:zgSDv+qnvS3Q9bkucWruq8tgs4MvRxZjhT6WlTic1so=
Enter 'yes' to accept, 'no' to abort; anything else to view:
"No carrier"
"No carrier"
"No carrier"
yes
"No carrier"
```
Any pointers are greatly appreciated @yuezk  (I can open a new issue if needed)

**Comment by rodricels** [comment]
Same problem here, openconnect requires "--servercert pin-sha256:xxxxxx" to work.

Is it possible to add parameters to vpn/openconnect service?

**Comment by yuezk** [comment]
> Same problem here, openconnect requires "--servercert pin-sha256:xxxxxx" to work.
> 
> Is it possible to add parameters to vpn/openconnect service?

Yes, you can open `/etc/systemd/system/gpservice.service` and append any arguments supported by the OpenConnect cli to `/usr/bin/gpservice`, including the `--servercert` argument.

https://github.com/yuezk/GlobalProtect-openconnect/blob/76a4977e9260a8123c39fc439944355ff6bff9aa/GPService/systemd/gpservice.service#L7

**Comment by rodricels** [comment]
Thanks @yuezk works like a charm!

**Comment by yuezk** [comment]
It should have been resolved in 1.3.0, as described in https://github.com/yuezk/GlobalProtect-openconnect#passing-the-custom-parameters-to-openconnect-cli

**Comment by Mart-Bogdan** [comment]
I would like to add, that you also need to restart service after changing /etc/gpservice/gp.conf  using command:
```
sudo systemctl restart gpservice.service
```
